sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/sources/RateStreamMicroBatchReader.scala

```diff
@@ -169,7 +170,7 @@ class RateStreamMicroBatchReader(options: DataSourceOptions, checkpointLocation:
     (0 until numPartitions).map { p =>
       new RateStreamMicroBatchInputPartition(
         p, numPartitions, rangeStart, rangeEnd, localStartTimeMs, relativeMsPerValue)
-        : InputPartition[Row]
+        : InputPartition[InternalRow]
```

Comment:
Is this needed? Doesn't RateStreamMicroBatchInputPartition implement InputPartition[InternalRow]?

Reply:
ditto

Reply:
This is fine since it isn't a cast, but it's generally better to check whether these are still necessary after refactoring.